src/autocomplete/shared/tables-unstable.ts

```diff
@@ -0,0 +1,99 @@
+import * as c3 from 'antlr4-c3';
+import {Lexer as LexerType, ParseTree, Parser as ParserType, TokenStream} from 'antlr4ng';
+import {createParser} from './query';
+import {
+    ColumnAliasSuggestion,
+    CursorPosition,
+    GetParseTree,
+    LexerConstructor,
+    ParserConstructor,
+    SqlAutocompleteResult,
+    SymbolTableVisitor,
+    Table,
+    TableContextSuggestion,
+} from './autocomplete-types';
+import {ColumnAliasSymbol, TableSymbol, getScope, getUniqueTableSuggestions} from './symbol-table';
+import {computeTokenContext} from './compute-token-position';
+
+interface ContextSuggestions {
+    tableContextSuggestion?: TableContextSuggestion;
+    suggestColumnAliases?: SqlAutocompleteResult['suggestColumnAliases'];
+}
+
+export function getTablesSuggestions<L extends LexerType, P extends ParserType>(
```

Comment:
I'd rename this file to `extended-tables.ts`
and this function to `getExtendedTableSuggestions`